File: frontend/src/types/street.ts
export interface Street {
  id: string;
  name: string;
  neighborhood: string | null;
  cep: string | null;
  is_active: boolean;
  created_at: string;
}

export interface StreetCreate {
  name: string;
  neighborhood?: string;
  cep?: string;
}

export interface StreetUpdate {
  name?: string;
  neighborhood?: string;
  cep?: string;
  is_active?: boolean;
}

export interface ClientSummary {
  id: string;
  name: string;
  phone: string | null;
}

export interface ClientInStreet {
  id: string;
  client_id: string;
  house_number: string | null;
  reference: string | null;
  display_order: number;
  client: ClientSummary;
}

export interface ReorderItem {
  client_street_id: string;
  display_order: number;
}

File: backend/app/schemas/street.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class StreetCreate(BaseModel):
    name: str
    neighborhood: Optional[str] = None
    cep: Optional[str] = None


class StreetUpdate(BaseModel):
    name: Optional[str] = None
    neighborhood: Optional[str] = None
    cep: Optional[str] = None
    is_active: Optional[bool] = None


class StreetResponse(BaseModel):
    id: uuid.UUID
    name: str
    neighborhood: Optional[str] = None
    cep: Optional[str] = None
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ClientSummary(BaseModel):
    id: uuid.UUID
    name: str
    phone: Optional[str] = None

    model_config = {"from_attributes": True}


class ClientInStreetResponse(BaseModel):
    id: uuid.UUID
    client_id: uuid.UUID
    house_number: Optional[str] = None
    reference: Optional[str] = None
    display_order: int
    client: ClientSummary

    model_config = {"from_attributes": True}


class ReorderItem(BaseModel):
    client_street_id: uuid.UUID
    display_order: int


class ReorderRequest(BaseModel):
    items: list[ReorderItem]
